lib/resources/firewalld.rb

```diff
@@ -0,0 +1,144 @@
+      result[0, result.length-1] == 'running'
+    end
+
+    def default_zone
+      # return: word associated with the name of the default zone
+      # example: 'public'
+      firewalld_command('firewall-cmd --get-default-zone')[0..-2]
+    end
+
+    def has_service_enabled_in_zone?(query_service, query_zone = default_zone)
+      firewalld_command("firewall-cmd --zone=#{query_zone} --query-service=#{query_service}")[0..-2] == 'yes'
+    end
+
+    def service_ports_enabled_in_zone(query_service, query_zone = default_zone)
+      # return: String of ports open
+      # example: ['22/tcp', '4722/tcp']
+      firewalld_command("firewall-cmd --zone=#{query_zone} --service=#{query_service} --get-ports --permanent").split(' ')
+    end
+
+    def service_protocols_enabled_in_zone(query_service, query_zone = default_zone)
+      # return: String of protocoals open
+      # example: ['icmp', 'ipv4', 'igmp']
+      firewalld_command("firewall-cmd --zone=#{query_zone} --service=#{query_service} --get-protocols --permanent").split(' ')
+    end
+
+    def has_port_enabled_in_zone?(query_port, query_zone = default_zone)
+      firewalld_command("firewall-cmd --zone=#{query_zone} --query-port=#{query_port}")[0..-2] == 'yes'
+    end
+
+    def has_rule_enabled?(rule, query_zone = default_zone)
+      rule = 'rule ' + rule
+      firewalld_command("firewall-cmd --zone=#{query_zone} --query-rich-rule=#{rule}")[0..-2] == 'yes'
+    end
+
+    private
+
+    def active_zones
+      # return syntax:
+      #   [default-zone-name]
+      #       interfaces: [open interfases]
+      #
+      # example:
+      #   public
+      #       interfaces: enp0s3
+      firewalld_command('firewall-cmd --get-active-zones')
+    end
+
+    def parse_active_zones(content)
+      # Split by every second line, which contains the zone and the interfaces.
+      content = content.split(/\n/).each_slice(2).map { |slice| slice.join("\n") }
+      content.map do |line|
+        parse_line(line)
+      end.compact
+    end
+
+    def parse_line(line)
+      zone = line.split("\n")[0]
+      {
+        'zone' => zone,
+        'interfaces' => line.split(':')[1].split(' '),
+        'services' =>   services_bound(zone),
+        'sources' =>    sources_bound(zone),
+      }
+    end
+
+    def sources_bound(query_zone)
+      # result: a list containing either an ip address or ip address with a mask, or a ipset or an ipset with the ipset prefix.
+      # example: ['192.168.0.4', '192.168.0.0/16', '2111:DB28:ABC:12::', '2111:db89:ab3d:0112::0/64']
+      firewalld_command("firewall-cmd --zone=#{query_zone} --list-sources").split(' ')
+    end
+
+    def services_bound(query_zone)
+      # result: a list of services bound to a zone.
+      # example: ['ssh', 'dhcpv6-client']
+      firewalld_command("firewall-cmd --zone=#{query_zone} --list-services").split(' ')
+    end
+
+    def firewalld_command(command)
```

Comment:
I think we discussed this last week, but just in case it slipped through the cracks...

I'd change this resource to look something like this:

```ruby
def firewalld_command(command_args)
  command_to_run = "firewall-cmd #{command_args}"
  result = inspec.command(command_to_run)
  # rest of method here...
end
```

... and then you don't have to repeat `firewall-cmd` in every call to it.  For example:

```ruby
firewalld_command("--zone=#{query_zone} --list-sources").split(' ')
```

This just eliminates a ton of repetition. Not a huge deal, but something I think makes sense to clean up.